src/Transport/Topology/Topologies/EndpointOrientedTopology.cs

```diff
@@ -90,6 +90,9 @@ private void InitializeContainer(SettingsHolder settings)
 
             var validationStrategyType = (Type)settings.Get(WellKnownConfigurationKeys.Topology.Addressing.Validation.Strategy);
             container.Register(validationStrategyType);
+
+            var publishersConfiguration = PublishersConfiguration.ConfigureWith(settings);
```

Comment:
Don't like that we have a constructor and at the same time `ConfigureWith()` method. It's not aligned. I could easily do ` var publishersConfiguration = new PublishersConfiguration(settings)`, how would I know that I'm missing an important dependency?

If you want ctor to be the creation point, remove `ConfigureWith`. If `ConfigureWith` is the construction point, then don't expose a constructor and rename `ConfigureWith` to represent what it does: `Create`.